src/shims/unix/fs.rs

```diff
@@ -1721,6 +1722,129 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.set_last_error(enotty)?;
         Ok(0)
     }
+
+    fn mkstemp(&mut self, template_op: &OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        use rand::seq::SliceRandom;
+
+        // POSIX defines the template string.
+        const TEMPFILE_TEMPLATE_STR: &str = "XXXXXX";
+        // POSIX defines the minimum number of attempts before failure.
+        // <https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/sysdeps/posix/tempname.c#L203>
+        const TEMPFILE_ATTEMPTS_MIN: u32 = 62 * 62 * 62;
+
+        let this = self.eval_context_mut();
+        this.assert_target_os_is_unix("mkstemp");
+
+        // Get the raw bytes from the template. This is passed to us from `std` as an `OsString`.
+        // These bytes represent a string in _target_ OS encoding, which may or may not match
+        // the _host_ OS encoding.
```

Comment:
```suggestion
        // Get the raw bytes from the template (as a byte slice, this is a string in the target).
```
There is no `OsString` in this block of code so I am rather confused by you bringing them up here.